nevergrad/optimization/optimizerlib.py

```diff
@@ -860,6 +860,45 @@ def _internal_provide_recommendation(self) -> tp.ArrayLike:
         return self.avg
 
 
+class _Rescaled(base.Optimizer):
+    """Proposes a version of a base optimizer which works at a different scale."""
+    def __init__(
+            self,
+            parametrization: IntOrParameter,
+            budget: tp.Optional[int] = None,
+            num_workers: int = 1,
+            base_optimizer: base.OptCls = CMA,
+            scale: tp.Optional[float] = None,
+    ) -> None:
+        super().__init__(parametrization, budget=budget, num_workers=num_workers)
+        self.base_optimizer = base_optimizer(parametrization, budget=budget, num_workers=num_workers)
+        self._subcandidates: tp.Dict[str, p.Parameter] = {}
+        if scale is None:
+            assert budget is not None, "Either scale or budget must be known in _Rescaled."
+            scale = np.sqrt(np.log(self.budget) / self.dimension)
+        self.scale = scale
+        assert self.scale != 0., "scale should be non-zero in Rescaler."
+
+    def rescale_candidate(self, candidate: p.Parameter, inverse: bool=False) -> p.Parameter:
+        data = candidate.get_standardized_data(reference=self.parametrization)
+        return (
+            self.parametrization.spawn_child().set_standardized_data(self.scale * data if not inverse
+            else (1. / self.scale) * data))
+
+    def _internal_ask_candidate(self) -> p.Parameter:
+        candidate = self.base_optimizer.ask()
+        sent_candidate = self.rescale_candidate(candidate)
+        self._subcandidates[sent_candidate.uid] = candidate
+        return sent_candidate
+
+    def _internal_tell_candidate(self, candidate: p.Parameter, loss: tp.FloatLoss) -> None:
+        self.base_optimizer.tell(self._subcandidates[candidate.uid], loss)
```

Comment:
Lets not keep the subcandidate in memory
```suggestion
        self.base_optimizer.tell(self._subcandidates.pop(candidate.uid), loss)
```

Reply:
Done, thx.